include/eve/module/core/regular/bit_ternary.hpp

```diff
@@ -0,0 +1,389 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch.hpp>
+#include <eve/traits/overload.hpp>
+#include <eve/module/core/decorator/core.hpp>
+#include <eve/module/core/regular/bit_and.hpp>
+#include <eve/module/core/regular/bit_andnot.hpp>
+#include <eve/module/core/regular/bit_or.hpp>
+#include <eve/module/core/regular/bit_notor.hpp>
+#include <eve/module/core/regular/bit_select.hpp>
+#include <eve/module/core/regular/bit_xor.hpp>
+#include <iostream>
+
+namespace eve
+{
+  template<typename Options>
+  struct bit_ternary_t : callable<bit_ternary_t, Options>
+  {
+    template<int K, value T0, value T1, value T2>
+    requires(eve::same_lanes_or_scalar<T0, T1, T2> && 0 <= K && K <= 255)
```

Comment:
Why not taking K as an std::uint8t directly ?

Reply:
done